ads/google/a4a/utils.js

```diff
@@ -106,16 +105,17 @@ export function extractGoogleAdCreativeAndSignature(
  * @param {!../../../extensions/amp-a4a/0.1/amp-a4a.AmpA4A} a4a
  * @param {string} baseUrl
  * @param {number} startTime
- * @param {number} slotNumber
  * @param {!Array<!./url-builder.QueryParameterDef>} queryParams
  * @param {!Array<!./url-builder.QueryParameterDef>} unboundedQueryParams
  * @param {(string|undefined)} clientId
  * @param {string} referrer
  * @return {string}
  */
 function buildAdUrl(
-    a4a, baseUrl, startTime, slotNumber, queryParams, unboundedQueryParams,
+    a4a, baseUrl, startTime, queryParams, unboundedQueryParams,
     clientId, referrer) {
+  const slotId = a4a.element.getAttribute('data-amp-slot-index');
+  const slotNumber = Number(slotId);
```

Comment:
This set by code in amp-ad, to a number. So I would prefer to simply remove the coerce.